src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java

```diff
@@ -78,6 +85,39 @@ public Response createDaaForDac(
     }
   }
 
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+  @Path("{daaId}/{libraryCardId}")
+  public Response createLibraryCardDaaRelation(
+      @Context UriInfo info,
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId,
+      @PathParam("libraryCardId") Integer libraryCardId) {
+    try {
+      DataAccessAgreement daa = daaService.findById(daaId);
+      User user = userService.findUserByEmail(authUser.getEmail());
+      int userInstitutionId = user.getInstitutionId();
+      LibraryCard libraryCard = libraryCardService.findLibraryCardById(libraryCardId);
+      int libraryCardInstitutionId = libraryCard.getInstitutionId();
```

Comment:
Instead of comparing the LC institution id, we should compare the researcher's institution id and the authed user's institution id in the case that the authed user is a Signing Official and not an Admin.